src/domain/authentication/authProvider.ts

```diff
@@ -67,4 +69,9 @@ const authProvider: AuthProvider = {
   },
 };
 
-export default authProvider;
+const authProviderWithRefreshAuth = addRefreshAuthToAuthProvider(
+  authProvider,
+  refreshAuth
+);
```

Comment:
So, this code refreshes the **expired** JWT token when **needed** and since it is linked to the authProvider, my assumption is that it does it when the silent renewal should be done, meaning when the token is expiring soonish.

The other function that refreshes it, is the "addRefreshAuthToDataProvider", which does this same thing when data is fetched from the API. This means that when the user is active, it should be refreshed along side handlign the queries.

Then a question: **Can the API token expire before the ID-token and authToken for the client?** This is ofcourse configured in the authroization server and not in the client, but does this React-Admin work so that the API-token renewal needs the dataProviderRefresher and the auth token for the client needs the authProviderRefresher?